jobs/src/test/scala/dcos/metronome/model/JobRunSpecValidatorTest.scala

```diff
@@ -0,0 +1,61 @@
+package dcos.metronome
+package model
+
+import org.scalatest.{ FunSuite, GivenWhenThen, Matchers }
+
+class JobRunSpecValidatorTest extends FunSuite with Matchers with GivenWhenThen {
+  test("Undefined file based secret is invalid") {
+    Given("a JobRunSpec with undefined, but referenced secrets")
+    val jobRunSpec = JobRunSpec(
+      cmd = Some("sleep 1000"),
+      volumes = Seq(SecretVolume("path", "undefined-vol-secret")))
+
+    When("The spec is validated")
+    val result = JobRunSpec.validJobRunSpec(jobRunSpec)
+
+    Then("a validation error is returned")
+    result.isFailure shouldBe true
```

Comment:
Wdyt about checking that the undefined secret is mentioned in the failure message here and below?

Reply:
Probably not really necessary.

Reply:
Yeah – We have some helper code in Marathon that parses those, but I didn't want to add this. No test in here really verifies the message matches the expectations.

Only checking for `contains` is probably easy enough, and has _some_ value.

Reply:
Yep, our Marathon "helpers" just check for `string.contains()` :D

Reply:
Yep, I added these as well. thanks for the suggestion :)